src/omero_zarr/raw_pixels.py

```diff
@@ -108,88 +75,117 @@ def planeGen() -> np.ndarray:
 
 
 def add_raw_image(
-    *,
-    planes: Iterator[np.ndarray],
-    size_z: int,
-    size_c: int,
-    size_t: int,
-    d_type: np.dtype,
+    image: omero.gateway.ImageWrapper,
     parent: Group,
     level_count: int,
-    cache_dir: Optional[str] = None,
-    cache_file_name_func: Optional[Callable[[int, int, int], str]] = None,
+    tile_width: Optional[int] = None,
+    tile_height: Optional[int] = None,
 ) -> List[str]:
-    """Adds the raw image pixel data as array to the given parent zarr group.
-    Optionally caches the pixel data in the given cache_dir directory.
-    Returns the number of resolution levels generated for the image.
 
-    planes: Generator returning planes in order of zct (whatever order
-            OMERO returns in its plane generator). Each plane must be a
-            numpy array with shape (size_y, sizex), or None to skip the
-            plane.
-    """
+    pixels = image.getPrimaryPixels()
+    d_type = image.getPixelsType()
+    size_c = image.getSizeC()
+    size_z = image.getSizeZ()
+    size_x = image.getSizeX()
+    size_y = image.getSizeY()
+    size_t = image.getSizeT()
+
+    if tile_width is None:
+        tile_width = 1024
+    if tile_height is None:
+        tile_height = 1024
+    tile_width = int(tile_width)
+    tile_height = int(tile_height)
+
+    print(
+        "sizes x: %s, y: %s, z: %s, c: %s, t: %s"
+        % (size_x, size_y, size_z, size_c, size_t)
+    )
+    print(f"tile_width: {tile_width}, tile_height: {tile_height}")
 
-    if cache_dir is not None:
-        cache = True
-    else:
-        cache = False
-        cache_dir = ""
+    chunk_count_x = math.ceil(size_x / tile_width)
+    chunk_count_y = math.ceil(size_y / tile_height)
 
+    # create "0" array if it doesn't exist
+    path = "0"
```

Comment:
Unrelated heads up that we might want to encapsulate these names in case we ever want to move to a `SHOULD` recommendation for the naming of the arrays that includes a prefix.